File: python/challenge-2025-10-30.py

def nth_prime(n):
    result = 2
    ith_prime = 1
    while ith_prime < n:
        result += 1
        if is_prime(result):
            ith_prime += 1
    return result

def is_prime(n):
    if n < 2:
        return False
    num_divisors = get_num_divisors(n)
    if num_divisors == 2:
        return True
    else:
        return False

def get_num_divisors(n):
    result = 0
    for i in range(1, n + 1):
        if n % i == 0:
            result += 1
    return result

File: javascript/challenge-2025-10-30.js
function nthPrime(n) {
  let result = 2;
  let ith_prime = 1;
  while (ith_prime < n)
  {
    result += 1;
    if (isPrime(result))
    {
      ith_prime += 1;
    }
  }
  return result;
}

function isPrime(n)
{
  if (n < 2)
  {
    return false;
  }
  let num_divisors = getNumDivisors(n);
  if (num_divisors == 2)
  {
    return true;
  }
  else
  {
    return false;
  }
}

function getNumDivisors(n)
{
  let result = 0;
  for (let i = 1; i < n + 1; i++)
  {
    if (n % i == 0)
    {
      result += 1;
    }
  }
  return result;
}
